python/config_settings/transition.bzl

```diff
@@ -0,0 +1,200 @@
+"""The transition module contains the rule definitions to wrap py_binary and py_test and transition
+them to the desired target platform.
+"""
+
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("//python:defs.bzl", _py_binary = "py_binary", _py_test = "py_test")
+
+def _transition_python_version_impl(_, attr):
+    return {"//python/config_settings:python_version": str(attr.python_version)}
+
+_transition_python_version = transition(
+    implementation = _transition_python_version_impl,
+    inputs = [],
+    outputs = ["//python/config_settings:python_version"],
+)
+
+def _transition_py_impl(ctx):
+    target = ctx.attr.target[0]
+    executable = ctx.actions.declare_file(ctx.attr.name + (".exe" if ctx.attr.is_windows else ""))
+    ctx.actions.symlink(
+        is_executable = True,
+        output = executable,
+        target_file = target[DefaultInfo].files_to_run.executable,
+    )
+    zipfile_symlink = None
+    if ctx.attr.is_windows:
+        zipfile = None
+        expected_target_path = target[DefaultInfo].files_to_run.executable.short_path[:-4] + ".zip"
+        for file in target[DefaultInfo].default_runfiles.files.to_list():
+            if file.short_path == expected_target_path:
+                zipfile = file
+        zipfile_symlink = ctx.actions.declare_file(ctx.attr.name + ".zip")
+        ctx.actions.symlink(
+            is_executable = True,
+            output = zipfile_symlink,
+            target_file = zipfile,
+        )
```

Comment:
So I understand what this logic is doing, but why is it doing it? The underlying zip file will be there. Is it just to make sure that "{name}.zip" shows up alongside the regular "{name}" instead of the underlying "_{name}.zip"?

That's fine, but why does it matter? Please add a brief comment to explain.

It's unfortunate we have to flatten here; I don't see a way to avoid that. A minor improvement would be to flatten the default outputs instead of runfiles (it'd be a smaller depset); though I'm not sure how much it'll actually help because, under the hood, building the zip requires flattening anyways.

Reply:
Addressed in 906bb6a.